### ISSUE TITLE
Can I build with SDL backend?

### ISSUE THREAD

**Issue by ziontee113** [G3bE/raylib-zig#143]
Hi. Thank you for the amazing work.

I heard that Raylib 5 started to support SDL backend instead of GLFW.
Is there a way that I can use Raylib with SDL backend through this Zig bindings?

Thank you very much!

**Comment by Not-Nik** [comment]
Not possible yet, because raylib's `build.zig` doesn't have an option to do that. I'll leave this open and create an issue over there.

**Comment by Gandalf-Le-Dev** [comment]
I can see that your PR was merged in raylib have you planned to update the zig binding with this change ?

**Comment by Not-Nik** [comment]
Fixed in a239736

**Comment by lepasona** [comment]
when i try this option the build fails because it can't find `SDL.h`
in zig.build:
```zig
const raylib_dep = b.dependency("raylib_zig", .{
    .target = target,
    .optimize = optimize,
    .platform = .sdl,
});
```
`$ zig build run`:
```
run
└─ run zig_raylib_test
   └─ zig build-exe zig_raylib_test Debug native
      └─ zig build-lib raylib Debug native 1 errors
/home/user/.cache/zig/p/raylib-5.5.0-whq8uExcNgQBBys4-PIIEqPuWO-MpfOJkwiM4Q1nLXVN/src/platforms/rcore_desktop_sdl.c:55:10: error: 'SDL.h' file not found
#include "SDL.h"                // SDL base library (window/rendered, input, timing... functionality)
         ^~~~~~~~
```
<details>
<summary>more of this log</summary>
<code><pre>/home/user/.cache/zig/p/raylib-5.5.0-whq8uExcNgQBBys4-PIIEqPuWO-MpfOJkwiM4Q1nLXVN/src/rcore.c:553:14: note: in file included from /home/user/.cache/zig/p/raylib-5.5.0-whq8uExcNgQBBys4-PIIEqPuWO-MpfOJkwiM4Q1nLXVN/src/rcore.c:553:
    #include "platforms/rcore_desktop_sdl.c"
             ^
error: warning(compilation): failed to delete '/home/user/zig-raylib-test/.zig-cache/tmp/981aa13a6bf66433-rcore.o.d': FileNotFound

error: the following command failed with 1 compilation errors:
/opt/zig-linux-x86_64-0.14.0/zig build-lib -lGLX -lX11 -lXcursor -lXext -lXfixes -lXi -lXinerama -lXrandr -lXrender -lX11 -lEGL -lwayland-client -lxkbcommon -cflags -std=gnu99 -D_GNU_SOURCE -DGL_SILENCE_DEPRECATION=199309L -fno-sanitize=undefined -DEXTERNAL_CONFIG_FLAGS -DSUPPORT_CAMERA_SYSTEM -DSUPPORT_GESTURES_SYSTEM -DSUPPORT_RPRAND_GENERATOR -DSUPPORT_MOUSE_GESTURES -DSUPPORT_SSH_KEYBOARD_RPI -DSUPPORT_WINMM_HIGHRES_TIMER -DSUPPORT_PARTIALBUSY_WAIT_LOOP -DSUPPORT_SCREEN_CAPTURE -DSUPPORT_GIF_RECORDING -DSUPPORT_COMPRESSION_API -DSUPPORT_AUTOMATION_EVENTS -DSUPPORT_CLIPBOARD_IMAGE -DSUPPORT_FILEFORMAT_BMP -DSUPPORT_FILEFORMAT_BMP -DSUPPORT_FILEFORMAT_PNG -DSUPPORT_FILEFORMAT_PNG -DSUPPORT_FILEFORMAT_JPG -DRL_SUPPORT_MESH_GPU_SKINNING -DSUPPORT_QUADS_DRAW_MODE -DSUPPORT_FILEFORMAT_PNG -DSUPPORT_FILEFORMAT_GIF -DSUPPORT_FILEFORMAT_QOI -DSUPPORT_FILEFORMAT_DDS -DSUPPORT_IMAGE_EXPORT -DSUPPORT_IMAGE_GENERATION -DSUPPORT_IMAGE_MANIPULATION -DSUPPORT_DEFAULT_FONT -DSUPPORT_FILEFORMAT_TTF -DSUPPORT_FILEFORMAT_FNT -DSUPPORT_TEXT_MANIPULATION -DSUPPORT_FONT_ATLAS_WHITE_REC -DSUPPORT_FILEFORMAT_OBJ -DSUPPORT_FILEFORMAT_MTL -DSUPPORT_FILEFORMAT_IQM -DSUPPORT_FILEFORMAT_GLTF -DSUPPORT_FILEFORMAT_VOX -DSUPPORT_FILEFORMAT_M3D -DSUPPORT_MESH_GENERATION -DSUPPORT_FILEFORMAT_WAV -DSUPPORT_FILEFORMAT_OGG -DSUPPORT_FILEFORMAT_MP3 -DSUPPORT_FILEFORMAT_QOA -DSUPPORT_FILEFORMAT_XM -DSUPPORT_FILEFORMAT_MOD -DSUPPORT_STANDARD_FILEIO -DSUPPORT_TRACELOG -DSUPPORT_MODULE_RSHAPES -DSUPPORT_MODULE_RTEXTURES -DSUPPORT_MODULE_RTEXT -DSUPPORT_MODULE_RMODELS -DSUPPORT_MODULE_RAUDIO -- /home/user/.cache/zig/p/raylib-5.5.0-whq8uExcNgQBBys4-PIIEqPuWO-MpfOJkwiM4Q1nLXVN/src/rcore.c /home/user/.cache/zig/p/raylib-5.5.0-whq8uExcNgQBBys4-PIIEqPuWO-MpfOJkwiM4Q1nLXVN/src/utils.c /home/user/.cache/zig/p/raylib-5.5.0-whq8uExcNgQBBys4-PIIEqPuWO-MpfOJkwiM4Q1nLXVN/src/rshapes.c /home/user/.cache/zig/p/raylib-5.5.0-whq8uExcNgQBBys4-PIIEqPuWO-MpfOJkwiM4Q1nLXVN/src/rtextures.c /home/user/.cache/zig/p/raylib-5.5.0-whq8uExcNgQBBys4-PIIEqPuWO-MpfOJkwiM4Q1nLXVN/src/rtext.c /home/user/.cache/zig/p/raylib-5.5.0-whq8uExcNgQBBys4-PIIEqPuWO-MpfOJkwiM4Q1nLXVN/src/rmodels.c /home/user/.cache/zig/p/raylib-5.5.0-whq8uExcNgQBBys4-PIIEqPuWO-MpfOJkwiM4Q1nLXVN/src/raudio.c /home/user/.cache/zig/p/raylib-5.5.0-whq8uExcNgQBBys4-PIIEqPuWO-MpfOJkwiM4Q1nLXVN/src/rglfw.c -cflags -- /home/user/zig-raylib-test/.zig-cache/o/77f84f608a26a5a26f43fd4c754b4be4/raygui.c -ODebug -I /home/user/.cache/zig/p/raylib-5.5.0-whq8uExcNgQBBys4-PIIEqPuWO-MpfOJkwiM4Q1nLXVN/src/external/glfw/include -I /home/user/.cache/zig/p/raylib-5.5.0-whq8uExcNgQBBys4-PIIEqPuWO-MpfOJkwiM4Q1nLXVN/src/platforms -I /home/user/zig-raylib-test/.zig-cache/o/89571ad95099ae322b7ce7ab0247575a -I /home/user/zig-raylib-test/.zig-cache/o/5b77fbef1e3de1166d29bc973ba5a3e1 -I /home/user/zig-raylib-test/.zig-cache/o/ad0bf5951068b32c181f90c3dd988378 -I /home/user/zig-raylib-test/.zig-cache/o/7178de341db36a07ebd3a7ac15b5adac -I /home/user/zig-raylib-test/.zig-cache/o/32315a5af1fb446226c736bc4a3d4ed1 -I /home/user/zig-raylib-test/.zig-cache/o/5f7163dfa0d9142a3dba474b84c28b33 -I /home/user/zig-raylib-test/.zig-cache/o/56857ca297c81700954014dce615d87a -I /home/user/zig-raylib-test/.zig-cache/o/72e1b4cb1f22c7578dac1364a85832bf -I /home/user/zig-raylib-test/.zig-cache/o/ec7fc80ab1c50227a77624703ed407eb -I /home/user/zig-raylib-test/.zig-cache/o/8151f00362ee26764aea7f1fdb412a4d -I /home/user/zig-raylib-test/.zig-cache/o/ec9326707b8780cb9bf0a619bbe5186c -I /home/user/zig-raylib-test/.zig-cache/o/12eab6766cd53ee0e71fe21827665aa4 -I /home/user/zig-raylib-test/.zig-cache/o/7d9dcc740496d7a85828ba36cd96f5b2 -I /home/user/zig-raylib-test/.zig-cache/o/de74033aef2c525e93dd00d9b617b761 -I /home/user/zig-raylib-test/.zig-cache/o/c44bf0230354dc1f40b3ccdb18a295d1 -I /home/user/zig-raylib-test/.zig-cache/o/cc9b32d22e1a8592b2ef101e506d593c -I /home/user/zig-raylib-test/.zig-cache/o/b675b0e6ddef1d124d86dea932f46aad -I /home/user/zig-raylib-test/.zig-cache/o/e672c08d2593f63f45296ee74d461d3f -I /home/user/.cache/zig/p/N-V-__8AAEp9UgBJ2n1eks3_3YZk3GCO1XOENazWaCO7ggM2/src -I /home/user/.cache/zig/p/raylib-5.5.0-whq8uExcNgQBBys4-PIIEqPuWO-MpfOJkwiM4Q1nLXVN/src -D_GLFW_X11= -D_GLFW_WAYLAND= -DPLATFORM_DESKTOP_SDL= -Mroot -lc --cache-dir /home/user/zig-raylib-test/.zig-cache --global-cache-dir /home/user/.cache/zig --name raylib -static --zig-lib-dir /opt/zig-linux-x86_64-0.14.0/lib/ --listen=- 
Build Summary: 20/26 steps succeeded; 1 failed
run transitive failure
└─ run zig_raylib_test transitive failure
   ├─ zig build-exe zig_raylib_test Debug native transitive failure
   │  ├─ zig build-lib raylib Debug native 1 errors
   │  └─ zig build-lib raylib Debug native (+38 more reused dependencies)
   └─ install transitive failure
      └─ install zig_raylib_test transitive failure
         └─ zig build-exe zig_raylib_test Debug native (+3 more reused dependencies)
error: the following build command failed with exit code 1:
/home/user/zig-raylib-test/.zig-cache/o/cb69b37cf9adbb7dff5b02569f90d0e5/build /opt/zig-linux-x86_64-0.14.0/zig /opt/zig-linux-x86_64-0.14.0/lib /home/user/zig-raylib-test /home/user/zig-raylib-test/.zig-cache /home/user/.cache/zig --seed 0x43be8b62 -Z3956dc7dcc9fd95c run
</pre>
</code>
</details>

i have SDL2 libraries installed system wide, and i can compile raylib itself with SDL2 following [these instructions](https://github.com/raysan5/raylib/discussions/3764#discussioncomment-8333464). i think `-lSDL2 -I/usr/include/SDL2 -D_REENTRANT` needs to be in the build command? but i'm not sure how to do that.
